packages/blob/src/put-multipart.ts

```diff
@@ -0,0 +1,358 @@
+// eslint-disable-next-line unicorn/prefer-node-protocol -- node:stream does not resolve correctly in browser and edge
+import { Readable } from 'stream';
+import type { BodyInit } from 'undici';
+import { fetch } from 'undici';
+import type { PutBlobApiResponse, PutBlobResult, PutBody } from './put';
+import { getApiUrl, validateBlobApiResponse } from './helpers';
+
+// Most browsers will cap requests at 6 concurrent uploads per domain (Vercel Blob API domain)
+const maxConcurrentUploads = 6;
+
+// 5MB is the minimum part size accepted by Vercel Blob
```

Comment:
can you explain this? wasn't 5Mb the maximum payload size accepted?
do we want to let the user decide the size?

Reply:
AWS (and Cloudflare) rules for multipart uploads are:
- each part must be at least 5MB
- only the last part can be less than 5MB

> do we want to let the user decide the size?
Yes I think we should indeed (I said the opposite in the PR but it's an easy addition). There would be two modes:
- multipart: true (automatic mode, we decide of the size of parts etc..)
- multipart: { ... } (manual mode, user provide options).